README.md
# tabulEIA

> tabulEIA Automatize the process of building species occurrence list for EIA reports.
Listed species are arearranged with their conservation and taxonomic status.

## website


> Package [webite](https://pecard.github.io/tabuleia/index.html).

## Disclaimer

This software is preliminary or provisional. No warranty, expressed or
implied, is made by the maintainer. Software releases constitute no
warranty. The software is provided *as is* on the condition that the
maintainer shall not be held liable or responsible for any damages
resulting from the authorized or unauthorized use of the software.

## What is tabulEIA?

tabulEIA provides an exhaustive list of species occurring at a given area of interest. 
It does so by retrieving pre-compiled data from many sources: fieldwork, eBird, iNaturalist, 
Flora-on, Atlases, Regional databases, EU reports. Lists are provided with taxonomic 
and conservation status and will be ready to be used in EIA report.
Data being retrieved from:

- Field inventories
- Atlases (mammals, birds, Amphibians, Reptiles, Orchids)
- Flora-on database
- eBird (5 minute min. census)
- iNaturalist (research grade records)
- Natura 2000 Report

## Installation

You can install the released version of tabulEIA from
[Github](https:://github.com/pecard/tabuleia) with:

``` r
library(devtools)
install_github("pecard/tabuleia")
```

## Contribute

You are welcome to contribute to the package development and improvement
by [filling an issue](https://github.com/pecard/tabuleia/issues) at the
package web page.

## Code of Conduct
  
  Please note that the tabulEIA project is released with a [Contributor Code of Conduct](https://contributor-covenant.org/version/2/0/CODE_OF_CONDUCT.html). 
  By contributing to this project, you agree to abide by its terms.
  
## Usage

Run a basic example:

``` r
library(tabulEIA)
library(tidyverse)
library(sf)
library(geotidy)

# Copy geometry directly from QGIS wkt
pol <- st_geomfromtext('Polygon ((-8.2522 37.8960, -8.2513 37.8960, -8.2513 37.8951,
                       -8.2522 37.8951, -8.2522 37.8960))')
pol <- st_as_sf(pol) %>% st_set_crs(4326)
utm_all = utm_id(grid = utm10k, roi = pol, buff = NULL, contiguity = 'queen')

# Flora
# inventories from field work
dados_campo <- read.delim('~path_to_my_field_data/fielddata.txt')

tflora <- tabulEIA::tabulFlora(fielddata = dados_campo, utm_ae = utm_all$ae, utm_q = utm_all$contig,
                               biblio = 'all')
# Avifauna
tave = tabulEIA::tabulAve(fielddata = NULL, utm_ae = utm_ae, utm_q = utm_contig, atlas = atlas_aves, ebird = ebird)

```


NEWS.md
# tabulEIA 0.1.0

* Added a `NEWS.md` file to track changes to the package.


R/utmID.R
#' Identify the UTM grid squares touching and around a region of interest (roi).
#' @description The \code{utm_id} function will identify the UTM grid codes
#' that touches and those in the vicinity (queen or rook) of roi.
#'
#' @param roi an spatial object sf, st, spatialpolygonsdataframe identifying
#' the roi.' It can be of any valid vector type (point, line, polygon or collection).
#'
#' @param buf numeric indication whether the roi should be buffered to a given distance (meter).
#' Default is NULL
#'
#' @param contiguity character identifying the contiguity search method "queen" or "rook".
#'
#' @param grid character indicating the UTM grid. Ensure you provide an 'UTM' column as
#' grid code.
#'
#' @return A vector with UTM codes for study site (ae) and contiguous grids.
#'
#' @details Make sure you have a topologically valid  vector. Using external files (like a shapefile)
#' will require the .prj otherwise identify it with the corresponding numeric epsg code.
#' Currently only rook and queen implemented (not bishop)
#'
#' @author Paulo E. Cardoso
#'
#' @import sf
#' @import sp
#' @import tidyverse
#' @export
#' @examples
#' \dontrun{
#' require(tabulEIA)
#' require(sf)
#' require(tidyverse)
#' # get UTM codes for your roi
#' d <- data.frame(a = 1)
#' d$geom <- "Polygon ((-8.2522 37.8960, -8.2513 37.8960, -8.2513 37.8951, -8.2522 37.8951, -8.2522 37.8960))"
#' ae <- sf::st_as_sf(d, wkt = "geom")
#' ae <- sf::st_set_crs(ae, 4326)
#' utm_all <- utm_id(grid = utm10k, roi = ae, buf = NULL, contiguity = "queen")
#' utm_ae <- utm_all$ae
#' utm_ae
#' utm_contig <- utm_all$contig
#' utm_contig
#' }
utm_id <- function(grid = utm10, roi = roi, buf = NULL, contiguity = "queen") {
  grid <- grid %>% dplyr::select("UTM")

  if(length(ae) > 1)  roi <- st_combine(roi)

  if (is.na(st_crs(roi))) stop("No projection provided for your roi")
  # check projection supplied
  if (sf::st_crs(roi)$epsg != 3763) {
    roi <- sf::st_transform(roi, 3763)
  }
  # buffer roi
  if (!is.null(buf)) {
    roi <- sf::st_buffer(roi, buf)
  }

  # Intersects
  if (any(class(roi) == "sf")) {
    utm_ae <- sf::st_join(roi, grid)[["UTM"]]
  }
  if (any(class(roi) == "character")) {
    utm_ae <- roi
  }
  # Contiguity
  if (contiguity == "queen") {
    utm_l <- list()
    for (i in utm_ae) {
      utm_l[[i]] <- grid %>%
        dplyr::mutate(
          nb_rook =
            as.numeric(ist_queen(.,
              b = grid %>%
                subset(UTM %in% i)
            ))
        ) %>%
        dplyr::filter(nb_rook == 1)
    }
    utm_c <- do.call(rbind, utm_l) %>%
      dplyr::distinct(UTM, .keep_all = T) %>%
      sf::st_set_geometry(NULL) %>%
      dplyr::select(UTM) %>%
      dplyr::filter(!UTM %in% utm_ae) %>%
      dplyr::pull(UTM)
  }
  if (contiguity == "rook") {
    utm_l <- list()
    for (i in utm_ae) {
      utm_l[[i]] <- grid %>%
        dplyr::mutate(
          nb_rook =
            as.numeric(ist_rook(.,
              b = grid %>%
                subset(UTM %in% i)
            ))
        ) %>%
        dplyr::filter(nb_rook == 1)
    }
    utm_c <- do.call(rbind, utm_l) %>%
      dplyr::distinct(UTM, .keep_all = T) %>%
      sf::st_set_geometry(NULL) %>%
      dplyr::select(UTM) %>%
      dplyr::filter(!UTM %in% utm_ae) %>%
      dplyr::pull(UTM)
  }
  utml <- list("ae" = utm_ae, "contig" = utm_c)
  return(utml)
}
